### PR TITLE
CuDNN v4 wrapper

### PR DESCRIPTION
**Pull request by mrakgr**
I have a [BN with convolutional net Mnist example](https://github.com/mrakgr/Spiral-V2/blob/master/Spiral%20Conv/batch_normalization_example.fsx) in my repo as well. I've tested it an it works.

Whether you want to remove the v3 cuDNN or leave it in, I will leave to your discretion.

As the CuDNN v5 RC just came out and it has the optimized Winograd convolutional kernels along with a bunch of other stuff like recurrent net functions, I'll do it as well while I am at it.

Edit: I've also fixed the descriptions for amin and amax.

### CODE REVIEW

**Comment by mrakgr** [comment]
This pull request now has both v4 and v5 wrappers. I finally did it. With each version cuDNN is really exploding with features. At this rate my own library will become just a front end for it. I would not mind it if you published this to Nuget as I will be using the v5 wrapper myself. I haven't tested the new functions yet, but I am too busy to get sidetracked again.

v4 was worth doing just for batch normalization and the optimized GRU and LSTM implementations in v5 will definitely come in handy in the future. Hopefully they will add BN to them as well. Just recently a paper came out showing how to make it work well.

Well, I think I will take a break with this, I earned it.

**Comment by kunzmi** [comment]
Finally, I could update the code...
I found something strange in LRNDescriptor.cs, where it got changed to RRNDescriptor at some places (some find/replace bugs?) I tried to get it back to LRNDescriptor, not sure if I got them all.
I also updated the Nuget package. Since this was the first package I built from my new Win10 machine and many things went wrong while building the package, I'm not sure if it actually works fine. Let me know if something is missing or is not functional!

Cheers,
Michael

**Comment by mrakgr** [comment]
I haven't checked it yet, but in v5 there are both LRNDescriptors for Local Response Normalization functions and RRNDescriptors for Recurrent Neural Net functions. I'll take a look at it later today.

**Comment by mrakgr** [comment]
I am taking a look at it and for some reason the `cudnnNanPropagation` enum seems to be missing. It should be in `CudaDNNTypes.cs`.

**Comment by mrakgr** [comment]
I case you haven't seen the edits, let me reiterate in this post that the cuDNN package is still not functional. I would suggest rebuilding it.

**Comment by kunzmi** [comment]
I pushed the worng package to NuGet, now the right version should be available as version "5.0.1".
In the repository's master branch the files should also be complete and correct; if not, please let me know.

-Michael

**Comment by mrakgr** [comment]
Everything seems to be in order. The only thing I regret doing is putting all the functions inside the RNN descriptor, but the main context file was getting crowded and I was starting to have difficulty finding my way around. Hopefully it should be fine.

Thank you for finding the time to get this done.
